core/templates/domain/statistics/PlaythroughObjectFactorySpec.ts

```diff
@@ -34,7 +34,8 @@ describe('Playthrough Object Factory', () => {
   });
 
   it('should create a new playthrough', () => {
-    var actions = [this.laof.createNew('AnswerSubmit', {}, 1)];
+    var actions = [this.laof.createNewAnswerSubmitAction(
```

Comment:
Hmm, shouldn't this fail to compile? Please add type information to `this.laof` so that the type-information is available.

Reply:
why should this fail?

Also I don't know how to add type in the variables with `this` in the spec file.

Reply:
It should fail because it's missing all of the customization args. And OK, just extract it as a typed `var` at the top of the describe block:

```typescript
describe('...', function() => {
  var laof: LearnerActionObjectFactory;

  beforeEach(() => {
    laof = TestBed.get(LearnerActionObjectFactory);
  });
```

Reply:
done